### PR TITLE
[webdriver] Convert "inline" helper function to pytest fixture

### PR DESCRIPTION
**Pull request by Hexcles**
`inline` calls some fixtures directly. This use has been deprecated and removed in pytest 4.0.

https://docs.pytest.org/en/latest/deprecations.html#calling-fixtures-directly

(This is blocking #26704 .)

### CODE REVIEW

**Comment by Hexcles** [comment]
Well let's hope the heuristics pick up all affected tests. If so, we can easily tell whether this is correct by comparing the results. Otherwise, I'll push this to a trigger branch to trigger a full run.

File: webdriver/tests/accept_alert/accept.py [hunk 1]
```diff
@@ -10,7 +9,7 @@ def accept_alert(session):
         "POST", "session/{session_id}/alert/accept".format(**vars(session)))
 
 
-def test_null_response_value(session, url):
+def test_null_response_value(session, url, inline):
```
Comment (whimboo) [inline webdriver/tests/accept_alert/accept.py:12]: While touching this line can you please remove the `url` fixture? It's not needed here.

File: webdriver/tests/back/back.py [hunk 2]
```diff
@@ -88,7 +87,7 @@ def test_fragments(session, url):
     assert session.url == test_pages[0]
 
 
-def test_history_pushstate(session, url):
+def test_history_pushstate(session, url, inline):
```
Comment (whimboo) [inline webdriver/tests/back/back.py:90]: Same here for `url`. Maybe scan through the changes and also correct others? Looks like a couple more exist.

File: webdriver/tests/element_clear/clear.py [hunk 3]
```diff
@@ -121,7 +120,7 @@ def test_keyboard_interactable(session):
                           ("time", "19:48", ""),
                           ("month", "2017-11", ""),
                           ("week", "2017-W52", "")])
-def test_input(session, add_event_listeners, tracked_events, type, value, default):
+def test_input(session, add_event_listeners, tracked_events, inline, type, value, default):
```
Comment (whimboo) [inline webdriver/tests/element_clear/clear.py:123]: Mind moving the global fixture before all other locally defined ones? So right after `session`. Applies to other cases as well.

**Review by whimboo — COMMENTED** [review]
I like that change. Just some random thoughts inline.

**Comment by jgraham** [comment]
I guess this is going to need a corresponding change for our internal tests that use this function.

**Comment by Hexcles** [comment]
Well luckily CI seems to have picked up everything: https://github.com/web-platform-tests/wpt/pull/26711/checks?check_run_id=1484321635

And miraculously I made only one mistake (`shadow_dom.py`).

I've addressed other comments as well (where possible, I scanned through the rest of the change to rectify similar problems).

**Comment by whimboo** [comment]
> I guess this is going to need a corresponding change for our internal tests that use this function.

Exactly. @jgraham, how would you suggest get this done? I assume a manual step for the downstream sync would be required?

**Comment by Hexcles** [comment]
Test results are looking good: https://github.com/web-platform-tests/wpt/pull/26711/checks?check_run_id=1487555559

The remaining differences seem irrelevant /  flaky to me.

**Comment by jgraham** [comment]
@whimboo as long as we know it's coming it shouldn't be too hard to make a patch that we could apply after the sync. It's not trivial to roll it into the sync push itself, but it could land on autoland ~right after.

**Comment by whimboo** [comment]
Ok, good. I will file a bug when this PR is ready.

@Hexcles mind rebasing your patch? We got at least an update for `/element_click/navigate.py`.

**Comment by Hexcles** [comment]
Merged from master just now

(BTW I think this PR should be squash-merged at the end.)

**Comment by Hexcles** [comment]
Any other comments?

The stability checks timed out because too many tests were affected. I believe this should be safe to merge.

**Comment by Hexcles** [comment]
Gentle ping. This PR is rather extensive and can quite easily go stale.

**Comment by Hexcles** [comment]
Ping again @whimboo @jgraham

**Review by jgraham — APPROVED** [review]

**Comment by whimboo** [comment]
Sorry, but I was away last week. Thanks a lot for getting this done @Hexcles!

**Pull request merged**